substrait_consumer/producers.py

```diff
@@ -138,6 +144,83 @@ def name(self):
         return "IsthmusProducer"
 
 
+class DataFusionProducer:
+    def __init__(self, db_connection=None):
+        self.ctx = SessionContext()
+        if db_connection is not None:
+            self.db_connection = db_connection
+        else:
+            self.db_connection = db_connection
+
+    def set_db_connection(self, db_connection):
+        self.db_connection = db_connection
+
+    def produce_substrait(self, sql_query: str, ibis_expr: str = None) -> str:
```

Comment:
@richtia just a thought, it seems like `produce_substrait` is a behavior which is common to all the producers. I would suggest to create a base class and extend from that. But of course it is out of scope for this PR.

Reply:
Agreed.  initially it was like that, but i didn't make too much sense because there was only 1-2 producers.  I'll create a backlog issues.